src/main/java/medic/gateway/alert/WebappPoller.java

```diff
@@ -61,9 +61,12 @@ public SimpleResponse pollWebapp() throws JSONException, MalformedURLException {
 		return response;
 	}
 
-	public Boolean moreMessagesToSend() {
-		return request.wtMessageCount() == MAX_WT_MESSAGES ||
-					request.statusUpdateCount() == MAX_WOM_STATUS_UPDATES;
+	public Boolean moreMessagesToSend(SimpleResponse lastResponse) {
+		if (lastResponse == null || lastResponse.isError()) {
```

Comment:
Why return false if `lastResponse` is null? When could this happen?

Reply:
Not sure if the library for requests might return `null` for any inner process that stop working and not handled properly.
It's not expensive check, still weight 1 in cost. But It's also easy to remove and I can remove it too from the other places that we doing it, we have `try - catch` to catch any null pointer exception.

Reply:
You definitely do not want to try/catch a null pointer - that's crazy. You could change it to run the loop again if it's null, but given that we don't know when it may be null it's impossible to know the right behaviour. I guess leave it as it is for safety?